app/Services/PrenoteService.php

```diff
@@ -194,4 +195,38 @@ private static function getPrenotesForPosition(int $positionId): Collection
             ->select('prenotes.key')
             ->pluck('prenotes.key');
     }
+
+    public static function createNewAirfieldPairingFromPrenote(
+        string $departureAirfield,
+        string $arrivalAirfield,
+        string $prenoteKey
+    ): void {
+        DB::transaction(
+            function () use ($departureAirfield, $arrivalAirfield, $prenoteKey) {
+                DB::table('airfield_pairing_prenotes')
+                    ->insert(
+                        [
+                            'origin_airfield_id' => Airfield::where('code', $departureAirfield)->firstOrFail()->id,
+                            'destination_airfield_id' => Airfield::where('code', $arrivalAirfield)->firstOrFail()->id,
+                            'prenote_id' => Prenote::where('key', $prenoteKey)->firstOrFail()->id,
+                            'created_at' => Carbon::now(),
+                        ]
+                    );
+            }
+        );
+    }
+
+    public static function deleteAirfieldPairingPrenoteForPair(
+        string $departureAirfield,
```

Comment:
Would it be eaiser to pass in the airfield models for these two airfields instead of having to find them in the body of the function?

Reply:
Was thinking in the name of keeping migrations short and clear, passing in the ICAO's would be nicer :)

Reply:
works for me!